server/src/main/java/org/jboss/as/server/deployment/module/ModuleSpecification.java

```diff
@@ -32,15 +36,17 @@ public class ModuleSpecification extends SimpleAttachable {
     /**
      * System dependencies are dependencies that are added automatically by the container.
      */
-    private final List<ModuleDependency> systemDependencies = new ArrayList<>();
+    private final Set<ModuleDependency> systemDependenciesSet = new LinkedHashSet<>();
 
-    /** Unique names of system dependencies, used to ensure only one system ModuleDependency per name is added. */
-    private final Set<ModuleIdentifier> systemDependenciesSet = new HashSet<>();
+    /**
+     * List view of {@link #systemDependenciesSet}.
+     */
```

Comment:
Optional: Add a `@Deprecated(forRemoval = true)` too? I understand we will have a follow-up to fully clean up the dual view of the dependencies kept in the Set and List. The same applies to `userDependencies`.

Reply:
Done